pkg/fleet/installer/setup/djm/emr.go

```diff
@@ -0,0 +1,182 @@
+		{
+			Key:   "DD_DATA_JOBS_ENABLED",
+			Value: "true",
+		},
+		{
+			Key:   "DD_INTEGRATIONS_ENABLED",
+			Value: "false",
+		},
+		{
+			Key:   "DD_DATA_JOBS_COMMAND_PATTERN",
+			Value: ".*org.apache.spark.deploy.*",
+		},
+		{
+			Key:   "DD_SPARK_APP_NAME_AS_SERVICE",
+			Value: "true",
+		},
+		{
+			Key:   "DD_INJECT_FORCE",
+			Value: "true",
+		},
+	}
+)
+
+type emrInstanceInfo struct {
+	IsMaster        bool   `json:"isMaster"`
+	InstanceGroupID string `json:"instanceGroupId"`
+}
+
+type cluster struct {
+	Name string `json:"Name"`
+}
+
+type emrResponse struct {
+	Cluster cluster `json:"cluster"`
+}
+
+type extraEmrInstanceInfo struct {
+	JobFlowID    string `json:"jobFlowId"`
+	ReleaseLabel string `json:"releaseLabel"`
+}
+
+// SetupEmr sets up the DJM environment on EMR
+func SetupEmr(s *common.Setup) error {
+
+	s.Packages.Install(common.DatadogAgentPackage, emrAgentVersion)
+	s.Packages.Install(common.DatadogAPMInjectPackage, emrInjectorVersion)
+	s.Packages.Install(common.DatadogAPMLibraryJavaPackage, emrJavaVersion)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("failed to get hostname: %w", err)
+	}
+	s.Config.DatadogYAML.Hostname = hostname
+	s.Config.DatadogYAML.DJM.Enabled = true
+
+	s.Config.DatadogYAML.ExpectedTagsDuration = "10m"
+	isMaster, clusterName, err := setupCommonEmrHostTags(s)
+	if err != nil {
+		return fmt.Errorf("failed to set tags: %w", err)
+	}
+	if isMaster {
+		setupEmrDriver(s, clusterName)
+	}
+	return nil
+}
+
+func setupCommonEmrHostTags(s *common.Setup) (bool, string, error) {
+
+	instanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "instance.json"))
+	if err != nil {
+		return false, "", fmt.Errorf("error reading instance file: %w", err)
+	}
+
+	var info emrInstanceInfo
+	if err = json.Unmarshal(instanceInfoRaw, &info); err != nil {
+		return false, "", fmt.Errorf("error umarshalling instance file: %w", err)
+	}
+
+	setHostTag(s, "instance_group_id", info.InstanceGroupID)
+	setHostTag(s, "is_master_node", strconv.FormatBool(info.IsMaster))
+	s.Span.SetTag("host_tag."+"is_master_node", info.IsMaster)
+
+	extraInstanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "extraInstanceData.json"))
+	if err != nil {
+		return info.IsMaster, "", fmt.Errorf("error reading extra instance data file: %w", err)
+	}
+
+	var extraInfo extraEmrInstanceInfo
+	if err = json.Unmarshal(extraInstanceInfoRaw, &extraInfo); err != nil {
+		return info.IsMaster, "", fmt.Errorf("error umarshalling extra instance data file: %w", err)
+	}
+	setHostTag(s, "job_flow_id", extraInfo.JobFlowID)
+	setHostTag(s, "cluster_id", extraInfo.JobFlowID)
+	s.Span.SetTag("emr_version", extraInfo.ReleaseLabel)
+
+	emrResponseRaw, err := executeCommandWithTimeout("aws", "emr", "describe-cluster", "--cluster-id", extraInfo.JobFlowID)
+	if err != nil {
+		return info.IsMaster, extraInfo.JobFlowID, fmt.Errorf("error describing emr cluster, using cluster id as name: %w", err)
```

Comment:
From my understanding, this is causing the install script to stop and fail by returning this error as the code calling it is doing
```go
isMaster, clusterName, err := setupCommonEmrHostTags(s)
if err != nil {
	return fmt.Errorf("failed to set tags: %w", err)
}
```

We should not fail the install (since it will skip the steps coming afterwards) when the `cluster_name` is not available since it requires AWS permissions that might not always be set

Reply:
I changed to return nil instead of the error, log a warning and kept the fallback to clusterId logic.